napari/layers/points/points.py

```diff
@@ -125,107 +121,93 @@ def __init__(
     ):
 
-        # Create a compound visual with the following four subvisuals:
-        # Lines: The lines of the interaction box used for highlights.
-        # Markers: The the outlines for each point used for highlights.
-        # Markers: The actual markers of each point.
-        visual = Compound([Line(), Markers(), Markers()])
-
         super().__init__(
-            visual,
-            name=name,
-            opacity=opacity,
-            blending=blending,
-            visible=visible,
+            name=name, opacity=opacity, blending=blending, visible=visible
         )
 
         self.events.add(
             mode=Event,
             size=Event,
+            edge_width=Event,
             face_color=Event,
             edge_color=Event,
             symbol=Event,
             n_dimensional=Event,
+            highlight=Event,
         )
         self._colors = get_color_names()
 
-        # Freeze refreshes
-        with self.freeze_refresh():
-            # Save the point coordinates
-            self._data = coords
-            self.dims.clip = False
-
-            # Save the point style params
-            self.symbol = symbol
-            self.n_dimensional = n_dimensional
-            self.edge_width = edge_width
-
-            self.sizes = size
-            self.edge_colors = list(
-                itertools.islice(
-                    ensure_iterable(edge_color, color=True), 0, len(self.data)
-                )
-            )
-            self.face_colors = list(
-                itertools.islice(
-                    ensure_iterable(face_color, color=True), 0, len(self.data)
-                )
-            )
+        # Save the point coordinates
+        self._data = coords
+        self.dims.clip = False
 
-            # The following point properties are for the new points that will
-            # be added. For any given property, if a list is passed to the
-            # constructor so each point gets its own value then the default
-            # value is used when adding new points
-            if np.isscalar(size):
-                self._size = size
-            else:
-                self._size = 10
+        # Save the point style params
+        self.symbol = symbol
+        self._n_dimensional = n_dimensional
+        self.edge_width = edge_width
+
+        # The following point properties are for the new points that will
+        # be added. For any given property, if a list is passed to the
+        # constructor so each point gets its own value then the default
+        # value is used when adding new points
+        if np.isscalar(size):
+            self._size = size
+        else:
+            self._size = 10
 
-            if type(edge_color) is str:
-                self._edge_color = edge_color
-            else:
-                self._edge_color = 'black'
+        if type(edge_color) is str:
+            self._edge_color = edge_color
+        else:
+            self._edge_color = 'black'
```

Comment:
This seems dangerous...? What if someone says `edge_color=(0, 0, 1)`? I suggest we use matplotlib's universal color handling here... (I believe I vendored it.)

Reply:
Having said that, I recognise that it was already like this, so we can leave this fix till later... But maybe add a TODO comment? Thanks!

Reply:
yeah, let's fix in another PR